jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -1201,6 +1207,56 @@ void java_object_factoryt::gen_nondet_init(
   }
 }
 
+/// Nondeterministically initializes an int i in the range min <= i <= max,
+/// where min is the integer represented by `min_value_expr` and max is the
+/// integer represented by `max_value_expr`.
+/// \param [out] assignments: A code block that the initializing assignments
+///   will be appended to.
+/// \param basename_prefix: Used for naming the newly created symbol.
+/// \param min_value_expr: Represents the minimum value for the integer.
+/// \param max_value_expr: Represents the maximum value for the integer.
+/// \return A symbol expression for the resulting integer.
+const symbol_exprt java_object_factoryt::gen_nondet_int_init(
+  code_blockt &assignments,
+  const std::string &basename_prefix,
+  const exprt &min_value_expr,
+  const exprt &max_value_expr)
+{
+  // Allocate a new symbol for the int
+  const symbolt &int_symbol = get_fresh_aux_symbol(
+    java_int_type(),
+    id2string(object_factory_parameters.function_id),
+    basename_prefix,
+    loc,
+    ID_java,
+    symbol_table);
+  symbols_created.push_back(&int_symbol);
+  const auto &int_symbol_expr = int_symbol.symbol_expr();
+
+  // Nondet-initialize it
+  gen_nondet_init(
```

Comment:
Hmm, not sure about this one. On the one hand, every function call to `gen_nondet_init` looks horrible because it takes so many parameters. On the other hand, it's nice to just rely on `gen_nondet_init` to generate _anything_ for you, and I think it clarifies its role in this file when it is used this way.
Also, for primitive types it also has a `add_source_location() = loc` operation in addition to what you wrote, so that would become more like four lines overall and not be much more readable than the current version.